client/test/unit/services/SessionService.spec.js

```diff
@@ -0,0 +1,100 @@
+/* jshint expr: true */
+/* global inject, expect */
+describe('SessionService', function () {
```

Comment:
Great use of unit tests 👍 

This should start becoming a requirement for core features as we become less dependent on end to end tests.